### DISCUSSION TITLE
Dead jobs don't leave the interface

### DISCUSSION THREAD

**Discussion by aquarelacs-rodrigo-araujo** [argoproj/argo-cd#6976] [General]
Hello, I have a problem with jobs in the argocd interface, I have jobs that have already been removed from kubernetes after a successful run for more than 15 days, only jobs remain in the interface even after being removed from k8s. 

![image](https://user-images.githubusercontent.com/83836037/129229017-a57ef88a-e1c8-480b-bbbc-f5141d3cc721.png)

**Comment by jessesuen** [comment]
This could be a stale cache problem. Is this resolved by visiting the cluster page and clicking the "Invalidate cache" button?

![image](https://user-images.githubusercontent.com/12677113/129786433-51860739-bdaf-497d-b675-a45304a2ba7c.png)

**Comment by aquarelacs-rodrigo-araujo** [comment]
I ran the cache invalidation process, but the jobs are still continuing on the interface.

**Reply by jessesuen** [reply]
Are you able to click on an individual job and show the live object YAML? This should fail if the object doesn't truly exist because we query this from K8s API server upon the panel load.

**Reply by aquarelacs-rodrigo-araujo** [reply]
![image](https://user-images.githubusercontent.com/83836037/130148501-37de25e4-fdaa-4f6c-94bc-4da1e36ddec5.png)

I don't see anything on the live manifests

**Comment by jessesuen** [comment]
These look like older completed CronJob jobs. I think you may have the misunderstanding that that these Job are supposed to disappear. CronJobs keep 3 successful and 1 failed by default. See:
https://kubernetes.io/docs/tasks/job/automated-tasks-with-cron-jobs/#jobs-history-limits

**Reply by aquarelacs-rodrigo-araujo** [reply]
@jessesuen, I already have it configured in cronjobs, the job is deleted in kubernetes by this function, it only continues in the argo interface.

![image](https://user-images.githubusercontent.com/83836037/130246129-a39051ab-e133-4452-b299-63d860352d74.png)
